js/views/inbox_view.js

```diff
@@ -258,12 +265,123 @@
       });
       this.listenTo(convoCollection, 'reset', removeAllConversations);
 
+      window.libsession
+        .getMessageQueue()
+        .events.addListener('success', this.handleMessageSentSuccess);
+
+      window.libsession
+        .getMessageQueue()
+        .events.addListener('fail', this.handleMessageSentFailure);
+
       Whisper.events.on('messageExpired', messageExpired);
       Whisper.events.on('userChanged', userChanged);
 
       // Finally, add it to the DOM
       this.$('.left-pane-placeholder').append(this.leftPaneView.el);
     },
+
+    async fetchHandleMessageSentData(m) {
+      // nobody is listening to this freshly fetched message .trigger calls
+      const tmpMsg = await window.Signal.Data.getMessageById(m.identifier, {
+        Message: Whisper.Message,
+      });
+
+      if (!tmpMsg) {
+        return null;
+      }
+
+      // find the corresponding conversation of this message
+      const conv = window.ConversationController.get(
+        tmpMsg.get('conversationId')
+      );
+
+      // then, find in this conversation the very same message
+      const msg = conv.messageCollection.models.find(
+        convMsg => convMsg.id === tmpMsg.id
+      );
+      return { conv, msg };
+    },
+
+    async handleMessageSentSuccess(m) {
```

Comment:
i think this majority of the logic in this function can be put in `messages.js`. The event listener can still be attached in this file.

E.g
```
const { msg } = fetchedData;
msg.onSentSuccess(m)
```

Reply:
fixed